### PR TITLE
Partially addresses ISSUE-542 (Part 1)

### PR DESCRIPTION
**Pull request by DiegoPino**
Part 1/3

Partially addresses https://github.com/Islandora-CLAW/CLAW/issues/542 

Handles: Drupal coding standards, docs, minor code fixes and removal of
old/never used code.

Part 2 will deal with Kernel and Unit tests.

### CODE REVIEW

File: src/Normalizer/EntityReferenceItemNormalizer.php [hunk 1]
```diff
@@ -51,7 +50,7 @@ public function __construct(LinkManagerInterface $link_manager, EntityResolverIn
    * {@inheritdoc}
    */
   public function normalize($field_item, $format = NULL, array $context = array()) {
-    /** @var $field_item \Drupal\Core\Field\FieldItemInterface */
+    /* @var $field_item \Drupal\Core\Field\FieldItemInterface */
```
Comment (dannylamb) [inline src/Normalizer/EntityReferenceItemNormalizer.php]: Cool.

File: src/Normalizer/FieldItemNormalizer.php [hunk 2]
```diff
@@ -132,15 +132,15 @@ protected function constructValue($data, $context) {
...
    *   The untranslated field item instance.
-   * @param $langcode
+   * @param string $langcode
    *   The langcode.
    *
    * @return \Drupal\Core\Field\FieldItemInterface
    *   The translated field item instance.
    */
-  protected function createTranslatedInstance(FieldItemInterface $item, $langcode) {
+  protected function createTranslatedInstance(FieldItemInterface $item, string $langcode) {
```
Comment (whikloj) [inline src/Normalizer/FieldItemNormalizer.php]: Type hinting on primitive datatypes is not supported in PHP 5
http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration

File: src/Normalizer/ContentEntityNormalizer.php [hunk 3]
```diff
@@ -170,7 +169,7 @@ public function normalize($entity, $format = NULL, array $context = array()) {
    *
    * @throws \Symfony\Component\Serializer\Exception\UnexpectedValueException
    */
-  public function denormalize($data, $class, $format = NULL, array $context = array()) {
+  public function denormalize(array $data, string $class, string $format = NULL, array $context = array()) {
```
Comment (whikloj) [inline src/Normalizer/ContentEntityNormalizer.php]: Type hinting on primitive datatypes is not supported in PHP 5
http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration

File: src/Normalizer/FieldNormalizer.php [hunk 4]
```diff
@@ -96,7 +96,7 @@ public function denormalize($data, $class, $format = NULL, array $context = arra
    * @return array
    *   The array of normalized field items.
    */
-  protected function normalizeFieldItems($field, $format, $context) {
+  protected function normalizeFieldItems(FieldItemListInterface $field, string $format, array $context) {
```
Comment (whikloj) [inline src/Normalizer/FieldNormalizer.php]: Type hinting on primitive datatypes is not supported in PHP 5
http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration

File: src/Normalizer/FieldItemNormalizer.php (hunk 2 above)
Comment (dannylamb) [inline src/Normalizer/FieldItemNormalizer.php]: I've always hated that about PHP5.

File: src/Normalizer/JsonldContextTrait.php [hunk 5]
```diff
@@ -1,45 +1,38 @@
...
+/**
    * JSON-LD context helper trait.
    *
    * @author Diego Pino Navarro
    *
    * @internal
    */
-trait JsonLdContextTrait
-{
+trait JsonLdContextTrait {
```
Comment (dannylamb) [inline src/Normalizer/JsonldContextTrait.php]: Does anything use this trait?  If there's no plans for it, I'd delete it.

Comment (DiegoPino) [inline src/Normalizer/JsonldContextTrait.php]: @dannylamb i just revived it for what you asked @Natkeeran to-do. Can remove of course.

Comment (dannylamb) [inline src/Normalizer/JsonldContextTrait.php]: @DiegoPino I'd remove it.  It's always in the history if we need it.

Comment (DiegoPino) [inline src/Normalizer/JsonldContextTrait.php]: OK. removing our first trait. 👍

File: src/Normalizer/FieldNormalizer.php (hunk 4 above)
Comment (DiegoPino) [inline src/Normalizer/FieldNormalizer.php]: Ok, will go for that and see how can I avoid `phpcs` complaining about those, I really forget about Php 5 sometimes. Thanks!

**Comment by DiegoPino** [comment]
@whikloj @dannylamb trait is gone, string is gone, re-used a parent phpdoc.

**Pull request merged**